WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m

```diff
@@ -1142,42 +1143,31 @@ - (void)richTextView:(WPRichTextView *)richTextView didReceiveLinkAction:(NSURL
         linkURL = [NSURL URLWithString:linkURL.path relativeToURL:url];
     }
 
-    WPWebViewController *controller = [WPWebViewController webViewControllerWithURL:linkURL];
-    [self.navigationController pushViewController:controller animated:YES];
+    WPWebViewController *webViewController = [WPWebViewController webViewControllerWithURL:linkURL];
+    UINavigationController *navController = [[UINavigationController alloc] initWithRootViewController:webViewController];
+    [self presentViewController:navController animated:YES completion:nil];
 }
 
 - (void)richTextView:(WPRichTextView *)richTextView didReceiveImageLinkAction:(WPRichTextImage *)imageControl
 {
-    UIViewController *controller;
-
-    if (imageControl.linkURL) {
-        NSString *url = [imageControl.linkURL absoluteString];
-
-        BOOL matched = NO;
-        NSArray *types = @[@".png", @".jpg", @".gif", @".jpeg"];
-        for (NSString *type in types) {
-            if (NSNotFound != [url rangeOfString:type].location) {
-                matched = YES;
-                break;
-            }
-        }
-
-        if (matched) {
-            controller = [[WPImageViewController alloc] initWithImage:imageControl.imageView.image andURL:imageControl.linkURL];
-        } else {
-            controller = [WPWebViewController webViewControllerWithURL:imageControl.linkURL];
-        }
+    UIViewController *controller = nil;
```

Comment:
An object variable declaration without an assignment is automatically assigned nil by the compiler.  Yay for llvm.